### PR TITLE
Libreswan packages fixes upgrade 3.29 and dnssec

### PR DESCRIPTION
**Pull request by antonyantony**
Edited by @diizzyy 

Maintainer: @lucize 
Compile tested: ?
Run tested: ?

Description:

### CODE REVIEW

**Comment by diizzyy** [comment]
Hi Antony,
Did you do any run tests and if so what platform/device?

**Comment by antonyantony** [comment]
> Hi Antony,
> Did you do any run tests and if so what platform/device?
Hi Daniel,
yes.  MIPS  TP-Link Archer C7, v2.  I am testing authentication using X509 Certificates. May be the initial tests was only using password/PSK. 

DISTRIB_RELEASE='SNAPSHOT'
DISTRIB_REVISION='r9916-4fb8a116d0'
DISTRIB_TARGET='ar71xx/generic'
DISTRIB_ARCH='mips_24kc'
DISTRIB_DESCRIPTION='OpenWrt SNAPSHOT r9916-4fb8a116d0'

Linux c7-32 4.14.113 #0 Sat May 11 09:01:44 2019 mips GNU/Linux  

thanks for looking into this.

**Comment by antonyantony** [comment]
I  am not sure why the checks fails. It compile for me. May be the unbound package's autodetection is looking for libevent in a worng place, on the host instead of build directory?  any ideas?

Makefile:197: recipe for target '/home/build/build_dir/build_dir/target-mips_24kc_musl/unbound-1.9.1/.configured_68b329da9893e34099c7d8ad5cb9c940' failed
make[2]: *** [/home/build/build_dir/build_dir/target-mips_24kc_musl/unbound-1.9.1/.configured_68b329da9893e34099c7d8ad5cb9c940] Error 1
make[2]: Leaving directory '/home/build/openwrt_packages/net/unbound'
time: package/feeds/packages/unbound/compile#13.64#5.88#18.63
package/Makefile:107: recipe for target 'package/feeds/packages/unbound/compile' failed

**Comment by antonyantony** [comment]
I figured out the error 
-       --with-libevent \
+       --with-libevent=$(STAGING_DIR)/usr  

I will try update the pull request.

**Comment by lucize** [comment]
by chance did you tried xfrm interface with it ? I understand that 4.19 does support it

**Comment by antonyantony** [comment]
> by chance did you tried xfrm interface with it ? I understand that 4.19 does support it

only on Fedora not on OpenWRT.  I haven't merged the xfrmi support yet.  It should work on openwrt too.

**Comment by antonyantony** [comment]
make sure unbound install unbound-event.h

**Comment by lucize** [comment]
LGTM you should bump PKG_RELEASE
tested x86

**Comment by antonyantony** [comment]
> LGTM you should bump PKG_RELEASE
bumped to 4

> tested x86
thanks

**Comment by lucize** [comment]
you are also touching net/unbound/Makefile and if  results binary changes then you must increase there also

**Comment by antonyantony** [comment]
> you are also touching net/unbound/Makefile and if results binary changes then you must increase there also
good point. I bumped that too.  -thanks.

**Comment by neheb** [comment]
ping @EricLuehrsen

**Comment by EricLuehrsen** [comment]
I wonder if we could make this a libunbound configuration and optional dependency. Thoughts... Unbound is large in ROM but it can still be installed on top of base OpenWrt say in a TL-WDR3600 (8/128). libevent lets it open more ports which is not necessary on such weak platforms. Actually default CONF file is one thread so --libpthreads could be removed (optional again), which actually shrinks it and improves performance if you only plan on one thread. ...thoughts.

**Comment by antonyantony** [comment]
@EricLuehrsen are you proposing to make it optional?
create a  config option for libunbound to enable - libunbound + libevent + libpthreads,  which  what libreswan need?
Let me know if this the direction you are thinking? 
I can give this a try. I am still learning the OpenWRT Makefile,  creating options and dependencies.
thanks for your feedback.

**Comment by EricLuehrsen** [comment]
Yes. Both libevent and libpthreads allow Unbound to perform better by having access to more resources. Yet, they incur initial overhead even if those resources aren't employed. An embedded distribution should have such options default off, but easy to enable as desired.

File: net/unbound/Makefile [hunk 1]
```diff
@@ -9,7 +9,7 @@ include $(TOPDIR)/rules.mk
 
 PKG_NAME:=unbound
 PKG_VERSION:=1.9.1
-PKG_RELEASE:=3
+PKG_RELEASE:=4
```
Comment (EricLuehrsen) [inline net/unbound/Makefile]: PR #9029 will like render this commit obsolete.

File: net/libreswan/Makefile [hunk 2]
```diff
@@ -39,7 +39,7 @@ $(call Package/libreswan/Default)
   SUBMENU:=VPN
   SECTION:=net
   CATEGORY:=Network
-  DEPENDS:= +libnss +librt +libevent2 +libevent2-pthreads +kmod-crypto-authenc \
+  DEPENDS:= +libldns +libnss +librt +libevent2 +libevent2-pthreads +libunbound +kmod-crypto-authenc \
```
Comment (EricLuehrsen) [inline net/libreswan/Makefile]: With PR #9029 if it stays as its, this dependency will be `+libunbound-heavy`

**Comment by antonyantony** [comment]
@lucize  also I have updated  libreswan 3.28, while waiting for libunbound get merged, should I send as one pull request or two? First get the fixes then go for 3.28? 

If you get chance test  branch libreswan-3.28-1 
https://github.com/antonyantony/packages

**Comment by lucize** [comment]
I think you can go directly to 3.28 + fixes and also if you add yourself as a maintainer would be better

**Comment by neheb** [comment]
This pull request is somewhat hard to follow with 10 commits. Can you squash them?

**Comment by lucize** [comment]
@antonyantony I've also tested 3.28 on x86

**Comment by lucize** [comment]
isn't this one better for xfrm ? https://github.com/libreswan/libreswan/commit/716f4b712724c6698469563e531dea3667507ceb

**Comment by antonyantony** [comment]
> isn't this one better for xfrm ? [libreswan/libreswan@716f4b7](https://github.com/libreswan/libreswan/commit/716f4b712724c6698469563e531dea3667507ceb)

It is probably better, when I tried there were conflicts in barf.in. which means I have pull more patches in.
I will try again.

**Comment by lucize** [comment]
@antonyantony whatabout CVE-2019-12312 will it be a 3.29 release shortly ?

**Comment by antonyantony** [comment]
> @antonyantony whatabout CVE-2019-12312 will it be a 3.29 release shortly ?
@lucize 
fixed in 3.28.  https://libreswan.org/security/CVE-2019-12312/CVE-2019-12312.txt

**Comment by lucize** [comment]
@neheb LGTM tested x86, this also fixes a CVE

File: net/libreswan/Makefile [hunk 3]
```diff
@@ -7,12 +7,13 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=libreswan
-PKG_VERSION:=3.27
-PKG_RELEASE:=3
+PKG_VERSION:=3.29
+PKG_RELEASE:=1
 
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
+PKG_SOURCE_URL:=https://github.com/libreswan/libreswan.git
```
Comment (neheb) [inline net/libreswan/Makefile]: This looks like a mistake. Use only one PKG_SOURCE_URL please.

Comment (lucize) [reply net/libreswan/Makefile]: Yes, it worked because the other one overwrote it

Comment (antonyantony) [reply net/libreswan/Makefile]: fixed. Thanks @neheb .

File: net/libreswan/Makefile [hunk 4]
```diff
@@ -39,9 +39,11 @@ $(call Package/libreswan/Default)
...
   SECTION:=net
   CATEGORY:=Network
-  DEPENDS:= +libnss +librt +libevent2 +libevent2-pthreads +kmod-crypto-authenc \
-	+kmod-crypto-hash +kmod-ipt-ipsec +iptables-mod-ipsec +ip-full +kmod-ip-vti \
-	+kmod-ipsec +kmod-ipsec4 +kmod-crypto-rng +IPV6:kmod-ipsec6 +IPV6:kmod-ip6-vti
+  DEPENDS:= +IPV6:kmod-ip6-vti +IPV6:kmod-ipsec6 +ip-full +iptables-mod-ipsec \
+	+kmod-crypto-aead +kmod-crypto-authenc +kmod-crypto-gcm \
+	+kmod-crypto-hash +kmod-crypto-rng +kmod-ip-vti +kmod-ipsec \
+	+kmod-ipsec4 +kmod-ipt-ipsec +libevent2 +libevent2-pthreads \
+	+libldns +libnss +librt +libunbound-heavy +nss-utils +nspr
```
Comment (neheb) [inline net/libreswan/Makefile]: libnss can be removed here. it gets picked up by nss-utils.

File: net/libreswan/Makefile [hunk 5]
```diff
@@ -64,7 +66,6 @@ endef
 TARGET_CFLAGS+= -Wno-error=format-nonliteral
```
Comment (neheb) [inline net/libreswan/Makefile]: No idea if this uses autotools or what. Does this support passing

```
--disable-warnings
```

to CONFIGURE_ARGS? I'd rather that be used.

Comment (antonyantony) [reply net/libreswan/Makefile]: it does not use autotools.

File: net/libreswan/Makefile (hunk 4 above)
Comment (antonyantony) [reply net/libreswan/Makefile]: removed libnss. and pushed an update. thanks @neheb

**Comment by lucize** [comment]
xfrmi was introduced but it has some bugs, and maybe for a later PR
https://github.com/openwrt/openwrt/commit/ae3e232b11ac7d76d6178ea308b3356d1baf33f0

**Comment by antonyantony** [comment]
> xfrmi was introduced but it has some bugs, and maybe for a later PR
> [openwrt/openwrt@ae3e232](https://github.com/openwrt/openwrt/commit/ae3e232b11ac7d76d6178ea308b3356d1baf33f0)

yes for later. It is good to see OpenWRT merged it. the main difference between strongswan and librewsan   woubd  if_in and if_out. Libreswan is only planning support one if_id, symetric. Strongswan plan to support incoming packets via one interface and outgoing via another.

File: net/libreswan/Makefile (hunk 5 above)
Comment (lucize) [reply net/libreswan/Makefile]: it was the best I knew at that moment, I didn't test it other ways

Comment (antonyantony) [reply net/libreswan/Makefile]: same feeling here.

Comment (lucize) [reply net/libreswan/Makefile]: I made a compile on x86 and it worked with without TARGET_CFLAGS+= -Wno-error=format-nonliteral, maybe the new version fixed the error, I think it was in the kernel module, don't remember exactly

Comment (lucize) [reply net/libreswan/Makefile]: @antonyantony ping on this

Comment (antonyantony) [reply net/libreswan/Makefile]: > I made a compile on x86 and it worked with without TARGET_CFLAGS+= -Wno-error=format-nonliteral, maybe the new version fixed the error, I think it was in the kernel module, don't remember exactly

thanks for the suggestion. Let me try and get back  in  few hours. May be it is not necessary anymore.  we should compile for 32bit  to be sure.

Comment (lucize) [reply net/libreswan/Makefile]: I've run tested it on an atom 32 bit target

**Comment by neheb** [comment]
Is this good to go?

**Comment by lucize** [comment]
@neheb yes

**Pull request merged**